### DISCUSSION TITLE
Dynamic class and hover error

### DISCUSSION THREAD

**Discussion by Hova25** [tailwindlabs/tailwindcss#10393] [Help]
Hello,

I am experiencing an issue with generating Tailwind CSS. 
I understand that Tailwind only generates styles when we write "bg-blue-500" for example, and it creates that class. 
I want to add the hover effect (hover:) to the class dynamically.

exemple : 
```
in angular : 

<div class="'hover:' + 'bg-blue-500'" >
  My div
</div>

in react jsx : 

const hover = "hover:"
const color = "bg-blue-500"
return (
  <div className={`${hover}${color}`} >
    My div
  </div>
)
```

Do you have a solution to make this ?

For Your Information : What I want to do is more complicated than that, for your understanding I put a simple example
For Your Information 2 : if i write : 
```
<div class="'hover:bg-blue-500" >
  My div
</div>
```

It works but i want write with variable for my project context

Thanks for your help

**Comment by adamwathan** [comment]
Hey! The entire class name needs to be present in the source files for Tailwind to find it. If you share a very concrete example of when/why you need this then I can probably suggest something that is more idiomatic Tailwind 👍

**Reply by ghost** [reply]
@adamwathan sorry to hijack this, but I came across this post while searching for solutions. Suppose I have a card component that has dynamic height values stored in 2 states (ex. height, hoverHeight) and I want to have the height change on hover. How could I achieve this?

Simple representation of current implementation:
```
const [height, setHeight] = useState()
const [hoverHeight, setHeight] = useState()

useEffect(() => {
  setHeight('h-[' + some variable dimensions here from ref objects + 'px]')
  setHoverHeight('hover:h-[' + some variable dimensions here from ref objects + 'px]')
}, [dependencies that can change the heights])

...

<div className={`${height} ${hoverHeight}`}>
  card contents
</div>

```

**Reply by wongjn** [reply]
You could consider using the `style` attribute and CSS variables, like:

```jsx
<div
  className="hover:!h-[--hover-height]"
  style={{
    height: `${height}px`,
    '--hover-height: `${hoverHeight}px`,
  }}
>
  card contents
</div>
```

**Reply by Hova25** [reply]
Hello,

I have find a solution, but it's not optimal...
You can add classes to safelist in tailwind config. But if you put to much classes the compilation will take too much time.

You can follow this link : https://tailwindcss.com/docs/content-configuration#using-regular-expressions

**Comment by aenial51** [comment]
hello
could you explain more if possible ?
[CheckMyRota]( https://www.checkmyrota.net/)

**Comment by Hova25** [comment]
Hello,

I work on a component library in typescript and angular 15.
I have created differents types to map tailwind classes, (text-color, bg-color, width, height, padding, margin, ...)
I want use variants (hover, focus, sm, md, ...) without remapped all properties  and just un a simple condition to add "hover:" or "focus:" or "md:"

Thanks for you responses
